### PR TITLE
added: internalize SDENSITY table

### PR DESCRIPTION
**Pull request by akva2**
To avoid deck usage downstream.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this opm-material=406 opm-simulators=2367 please

File: opm/parser/eclipse/EclipseState/Tables/SolventDensityTable.hpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*
+  Copyright (C) 2015 Statoil ASA.
+  2015 IRIS AS
```
Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/Tables/SolventDensityTable.hpp]: The copy & paste demon strikes ... :-)

File: opm/parser/eclipse/EclipseState/Tables/TableManager.hpp [hunk 2]
```diff
@@ -283,6 +285,21 @@ namespace Opm {
         }
 
 
+        template <class TableType>
```
Comment (joakim-hove) [inline opm/parser/eclipse/EclipseState/Tables/TableManager.hpp]: Does this really need to be templated?

File: src/opm/parser/eclipse/EclipseState/Tables/SolventDensityTable.cpp [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        SolventDensityTable::SolventDensityTable()
+        {
+        }
+
+        SolventDensityTable::SolventDensityTable(const std::vector<double>& tableValues)
+            : m_tableValues(tableValues)
+        {
+        }
+
+        void SolventDensityTable::init(const Opm::DeckRecord& record )
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/EclipseState/Tables/SolventDensityTable.cpp]: Hmmm - I wonder who devised that funny `::init( )` pattern?

File: opm/parser/eclipse/EclipseState/Tables/TableManager.hpp (hunk 2 above)
Comment (akva2) [reply opm/parser/eclipse/EclipseState/Tables/TableManager.hpp]: not really, i just blindly followed the code i monkey'd (the brine density).

File: opm/parser/eclipse/EclipseState/Tables/SolventDensityTable.hpp (hunk 1 above)
Comment (akva2) [reply opm/parser/eclipse/EclipseState/Tables/SolventDensityTable.hpp]: fixed.

File: src/opm/parser/eclipse/EclipseState/Tables/SolventDensityTable.cpp (hunk 3 above)
Comment (akva2) [reply src/opm/parser/eclipse/EclipseState/Tables/SolventDensityTable.cpp]: you want me to change it? again i just followed the brine density.

**Comment by akva2** [comment]
jenkins build this opm-material=406 opm-simulators=2367 please

File: src/opm/parser/eclipse/EclipseState/Tables/SolventDensityTable.cpp (hunk 3 above)
Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Tables/SolventDensityTable.cpp]: No that is fine - I know you just followed the existing pattern; and I assume I am responsible for the original pattern :-)

**Comment by akva2** [comment]
since this is approved here, and the downstreams are fairly trivial, i will merge this patchset.

**Pull request merged**